nlu.py

```diff
@@ -20,6 +20,34 @@ def _get_commands(text_sequence: [str]) -> set:
     return possible_commands
 
 
+def _consine_similarity(question: [str], text_sequence: [str]) -> float:
+    words = set(question+text_sequence)
+    question_frequency = _get_frequency(words, question)
+    text_frequency = _get_frequency(words, text_sequence)
+    return _cosine_theta(question_frequency, text_frequency)
+
+
+def _cosine_theta(a: [int], b: [int]):
+    denominator = (sum(pow(n, 2) for n in a) ** 0.5)*(sum(pow(n, 2) for n in b) ** 0.5)
+    numerator = sum(a[i] * b[i] for i in range(len(a)))
+    return numerator/denominator
+
+
+def _get_frequency(source: set, text_sequence: [str]) -> [int]:
+    frequency = {}
+    for text in text_sequence:
+        if text in source:
+            value = frequency.get(text)
+            if value is None:
+                value = 1
+            else:
+                value += 1
+            frequency[text] = value
+        else:
+            frequency[text] = 0
+    return list(frequency.values())
```

Comment:
`values()` return a `list` doesn't it?

Reply:
actaully dict.values() return view objects in python3.
Ref: [official document](https://docs.python.org/3/library/stdtypes.html#dict-views), [stackoverflow](https://stackoverflow.com/questions/16228248/how-can-i-get-list-of-values-from-dict)